lib/Basic/SourceLoc.cpp

```diff
@@ -408,6 +408,66 @@ SourceManager::getGeneratedSourceInfo(unsigned bufferID) const {
   return known->second;
 }
 
+ArrayRef<CharSourceRange> IfConfigRangeInfo::getRangesWithoutActiveBody(
+    SourceManager &SM, SmallVectorImpl<CharSourceRange> &scratch) const {
+  assert(scratch.empty());
+  if (ActiveBodyRange.isValid()) {
+    // Split the whole range by the active range.
+    scratch.emplace_back(SM, WholeRange.getStart(), ActiveBodyRange.getStart());
+    scratch.emplace_back(SM, ActiveBodyRange.getEnd(), WholeRange.getEnd());
```

Comment:
Not sure if this could cause issue but should we only add the ranges if they are not empty?

Reply:
`WholeRange` should definitely never be empty, I believe `ActiveBodyRange` _could_ be empty, but only in invalid code (we error on e.g `#if true#endif`), but it shouldn't make a difference here, i.e we want the entire range in that case anyway. I'll at least add an assert for `WholeRange` though.

Reply:
But given https://github.com/apple/swift/pull/69782#discussion_r1392645561, looks like this will never come up in practice anyway.

Reply:
Ah right, I don't think that can happen, since the active range never includes the syntax of the `#if` and the whole range does. I guess we could add a check to be defensive though

Reply:
Oh heh only just saw your latest comment :)